examples/tv-app/linux/include/application-basic/ApplicationBasicManager.cpp

```diff
@@ -0,0 +1,117 @@
+{
+    if (strcmp(application->vendorName, "\0") != 0)
+    {
+        EmberAfStatus vendorNameStatus =
+            emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_APPLICATION_VENDOR_NAME_ATTRIBUTE_ID,
+                                        (uint8_t *) &application->vendorName, ZCL_CHAR_STRING_ATTRIBUTE_TYPE);
+        assert(vendorNameStatus == EMBER_ZCL_STATUS_SUCCESS);
+    }
+
+    if (application->vendorId != 0)
+    {
+        EmberAfStatus vendorIdStatus =
+            emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_APPLICATION_VENDOR_ID_ATTRIBUTE_ID,
+                                        (uint8_t *) &application->vendorId, ZCL_INT16U_ATTRIBUTE_TYPE);
+        assert(vendorIdStatus == EMBER_ZCL_STATUS_SUCCESS);
+    }
+
+    if (strcmp(application->name, "\0"))
+    {
+        EmberAfStatus nameStatus =
+            emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_APPLICATION_NAME_ATTRIBUTE_ID,
+                                        (uint8_t *) &application->name, ZCL_CHAR_STRING_ATTRIBUTE_TYPE);
+        assert(nameStatus == EMBER_ZCL_STATUS_SUCCESS);
+    }
+
+    if (application->productId != 0)
+    {
+        EmberAfStatus productIdStatus =
+            emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_APPLICATION_PRODUCT_ID_ATTRIBUTE_ID,
+                                        (uint8_t *) &application->productId, ZCL_INT16U_ATTRIBUTE_TYPE);
+        assert(productIdStatus == EMBER_ZCL_STATUS_SUCCESS);
+    }
+
+    if (strcmp(application->id, "\0"))
+    {
+        EmberAfStatus idStatus =
+            emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_APPLICATION_ID_ATTRIBUTE_ID,
+                                        (uint8_t *) &application->id, ZCL_CHAR_STRING_ATTRIBUTE_TYPE);
+        assert(idStatus == EMBER_ZCL_STATUS_SUCCESS);
+    }
+
+    if (application->catalogVendorId != 0)
+    {
+        EmberAfStatus catalogVendorIdStatus =
+            emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_CATALOG_VENDOR_ID_ATTRIBUTE_ID,
+                                        (uint8_t *) &application->catalogVendorId, ZCL_INT16U_ATTRIBUTE_TYPE);
+        assert(catalogVendorIdStatus == EMBER_ZCL_STATUS_SUCCESS);
+    }
+
+    EmberAfStatus applicationStatus =
+        emberAfWriteServerAttribute(endpoint, ZCL_APPLICATION_BASIC_CLUSTER_ID, ZCL_APPLICATION_STATUS_ATTRIBUTE_ID,
+                                    (uint8_t *) &application->status, ZCL_ENUM8_ATTRIBUTE_TYPE);
+    assert(applicationStatus == EMBER_ZCL_STATUS_SUCCESS);
+}
+
+Application ApplicationBasicManager::getApplicationForEndpoint(chip::EndpointId endpoint)
+{
+    Application app = {};
+    uint16_t size   = static_cast<uint16_t>(sizeof(app.name));
+
+    string section = "endpoint" + std::to_string(endpoint);
+    assert(es->get(section, "name", app.name, size) == CHIP_NO_ERROR);
```

Comment:
assert will get compiled out in production code.

I believe we want this to have sideffects, so you cannot use assert.

Reply:
Updated.